### PR TITLE
Move most of the magic to the mock_api_obj fixture

### PR DESCRIPTION
**Pull request by Cadair**
# Description

This is a few small improvements to the ExternalAPIMockServer fixtures.

* Move all the setup into the non-running fixture 
* Add a context manager to run the server (as an alternative to `run_test`).

## Status
**READY**

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: opsdroid/testing/fixtures.py [hunk 1]
```diff
@@ -76,17 +76,20 @@ def opsdroid() -> OpsDroid:
...
 
     This fixture returns an instance that hasn't been started, allowing you to
     add routes to the server before it is started.
     There are a few options of how to start the server once this fixture has
     been used:
 
-        * Use `:meth:opsdroid.testing.ExternalAPIMockServer.run_test`
         * Use the ``mock_api`` fixture in the test *after* this fixture has
           been called to setup a route.
+        * Use :meth:`opsdroid.testing.ExternalAPIMockServer.run_test`
```
Comment (Cadair) [inline opsdroid/testing/fixtures.py:89]: Should we just remove `run_test` at this point, I personally don't think it's a pattern we should support.

**Review by jacobtomlinson — APPROVED** [review]
Woo context managers

File: opsdroid/testing/fixtures.py (hunk 1 above)
Comment (FabioRosado) [reply opsdroid/testing/fixtures.py:89]: If we have the context manager, we probably don't need this one 🤔

**Review by FabioRosado — APPROVED** [review]
Context Managers are the best thing ever haha great work on this! If we remove `run_test` we need to update the docstring on `ExternalAPIMockServer` since the example contains `run_test`

**Comment by Cadair** [comment]
Pushed a new commit removing run_test and updating the name of the context manager to just `running` as I think it reads better `... with mock_api.running():`

**Comment by Cadair** [comment]
I bumped the codecov threshold now we aren't holding onto 100%, I have found with sunpy that the default of 0.1% can trigger just when you remove some code, 0.2% seems a little more stable.

**Pull request merged**